pkg/controllers/provisioning/scheduling/topologygroup.go

```diff
@@ -15,54 +15,283 @@ limitations under the License.
 package scheduling
 
 import (
+	"fmt"
 	"math"
 
+	"github.com/mitchellh/hashstructure/v2"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/util/sets"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	utilsets "k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/utils/sets"
+)
+
+type TopologyType byte
+
+const (
+	TopologyTypeSpread TopologyType = iota
```

Comment:
Just a thought, this might be slightly simpler to model with a string, since you can name all of the types and avoid the extra .String() complexity. I'm not sure the performance difference is worth the extra wrinkle.